### ISSUE TITLE
Compilation of .cu (CUDA) files fail when linking against AWS SDK

### ISSUE THREAD

**Issue by lablabla** [aws/aws-sdk-cpp#1353]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) and [API reference](http://sdk.amazonaws.com/cpp/api/LATEST/index.html)
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-cpp/issues) and didn't find any solution
  
**Describe the bug**
Compiling a project that uses both `AWS sdk` and `CUDA` causes compilation error `nvcc fatal   : Unknown option 'pthread'`

**SDK version number**
1.7.111
CUDA version 10.0.130
CMake 3.14

**Platform/OS/Hardware/Device**
What are you running the sdk on?
Ubuntu 16.04 

**To Reproduce (observed behavior)**
Steps to reproduce the behavior (please share code)

```
CMakeLists.txt
----------------
cmake_minimum_required (VERSION 3.12)
enable_language(CUDA)

find_package(AWSSDK REQUIRED COMPONENTS s3)


add_executable(exec main.cpp cuda.cu)

target_link_libraries(exec
        ${AWSSDK_LINK_LIBRARIES})
```
```
main.cpp
-----------
#include <iostream>

int main()
{
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
```
`cuda.cu` is empty

**Expected behavior**
Compilation should succeed

**Logs/output**
```
Scanning dependencies of target exec
[ 25%] Building CXX object CMakeFiles/exec.dir/main.cpp.o
[ 50%] Building CUDA object CMakeFiles/exec.dir/cuda.cu.o
nvcc fatal   : Unknown option 'pthread'
CMakeFiles/exec.dir/build.make:75: recipe for target 'CMakeFiles/exec.dir/cuda.cu.o' failed
make[3]: *** [CMakeFiles/exec.dir/cuda.cu.o] Error 1
```

**Additional context**
After digging around we saw that linking `${AWSSDK_LINK_LIBRARIES}` adds `-pthread` to the `CUDA_FLAGS`

**Comment by lablabla** [comment]
This seems to maybe be a cmake issue.
If I change the `enable_language(CUDA)` to the old `find_package(CUDA)`, it works find

closing

**Comment by lablabla** [comment]
Well,
After checking it more with CMake, looks like the problem might come from AWS after all

```
cmake_minimum_required (VERSION 3.12)
enable_language(CUDA) 

#find_package(AWSSDK REQUIRED COMPONENTS s3)

set (THREADS_PREFER_PTHREAD_FLAG ON)
find_package (Threads REQUIRED)

add_executable(exec main.cpp cuda.cu)

target_link_libraries(exec
        ${AWSSDK_LINK_LIBRARIES}
        ${CMAKE_THREAD_LIBS_INIT})
```
Enabling the `find_package(AWSSDK)` line will cause the `-pthread` flag to be added to the CUDA flags. finding pthread the normal way won't affect the CUDA flags

**Comment by KaibaLopez** [comment]
Hi @lablabla ,
Sorry for the wait, I was finally able to get around to test this, but it is not showing me the error you're talking about.
I have both a simple cuda sample from their [even easier introduccion](https://devblogs.nvidia.com/even-easier-introduction-cuda/) and a simple s3 upload object sample working on the same project compiling with no problems.

I used your same CMakeLists.txt too and was not able to reproduce your error.
My setup is a bit different than yours though:
```
cmake version 1.13 (shouldn't matter as you tested previous and newer versions)
Cuda compilation tools, release 10.0, V10.0.130
SDK Version: 1.7.314
Ubuntu 18.04
gcc 7.4
```
Not sure, but could you try upgrading the sdk to the newer versions see if by some off chance we inadvertently fixed this?

**Comment by justinpa-asg** [comment]
justin.palmer@asg.com is no longer a valid e-mail address.  Please contact Tim Wake at tim.wake@asg.com with any questions, comments, or concerns.

Diese justin.palmer@asg.com ist keine gultige Email-Adresse mehr. Bitte setzen Sie sich mit Tim Wake an folgende Email-Adresse tim.wake@asg.com in Verbindung, falls Sie etwaige Fragen, Kommentare oder Bedenken diesbezuglich haben.

Cette justin.palmer@asg.com n'est plus une adresse de courriel valable. Veuillez contacter Tim Wake a l'adresse de courriel suivante tim.wake@asg.com si vous avez des questions, des commentaires ou des preoccupations quelconques.

From: noreply@github.com
Sent: Monday, April 20 2020 6:32 PM
To: aws-sdk-cpp@noreply.github.com
CC: subscribed@noreply.github.com
Subject: Re: [aws/aws-sdk-cpp] Compilation of .cu (CUDA) files fail when linking against AWS SDK (#1353)


Closed #1353<http://TOC.ASG.COM:8080/?dmVyPTEuMDAxJiZkYWZlMDJkZTJlMzc4ZDM1YT01RTlERUFDRV80OTUyMl8zMDU0XzEmJjE5NzVjNDMyNGQzMjEzYj0xMjMyJiZ1cmw9aHR0cHMlM0ElMkYlMkZnaXRodWIlMkVjb20lMkZhd3MlMkZhd3Mtc2RrLWNwcCUyRmlzc3VlcyUyRjEzNTM=>.

**Comment by Tumb1eweed** [comment]
met the same problem, when add enable_language(CUDA), it automatically adds "-Os" to CUDA_FLAGS
